operators/src/source/gdal_source.rs

```diff
@@ -0,0 +1,456 @@
+        };
+
+        let time_interval_provider = vec![TimeInterval::new_unchecked(1.into(),2.into()), TimeInterval::new_unchecked(2.into(), 3.into())];
+        
+        let gdal_params = GdalSourceParameters{
+            base_path: "".into(),
+            file_name_with_time_placeholder: "".into(),
+            time_format: "".into(),
+            tick: None,
+            channel: None,
+        };
+
+        let gdal_source = GdalSource {
+            time_interval_provider,
+            grid_tile_provider,
+            gdal_params
+        };
+
+
+        let vres: Vec<_> = gdal_source.time_tile_iter().collect();
+        
+        assert_eq!(vres[0], (TimeInterval::new_unchecked(1.into(),2.into()), TileInformation::new(global_size_in_tiles.into(), (0, 0).into(), (0, 0).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[1], (TimeInterval::new_unchecked(1.into(),2.into()), TileInformation::new(global_size_in_tiles.into(), (0, 1).into(), (0, 1000).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[6], (TimeInterval::new_unchecked(1.into(),2.into()), TileInformation::new(global_size_in_tiles.into(), (1, 0).into(), (1000, 0).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[7], (TimeInterval::new_unchecked(1.into(),2.into()), TileInformation::new(global_size_in_tiles.into(), (1, 1).into(), (1000, 1000).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[34], (TimeInterval::new_unchecked(1.into(),2.into()), TileInformation::new(global_size_in_tiles.into(), (5, 4).into(), (5000, 4000).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[35], (TimeInterval::new_unchecked(1.into(),2.into()), TileInformation::new(global_size_in_tiles.into(), (5, 5).into(), (5000, 5000).into(), tile_size_in_pixels.into())));
+
+        assert_eq!(vres[36], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (0, 0).into(), (0, 0).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[36+1], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (0, 1).into(), (0, 1000).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[36+6], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (1, 0).into(), (1000, 0).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[36+7], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (1, 1).into(), (1000, 1000).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[36+34], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (5, 4).into(), (5000, 4000).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[36+35], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (5, 5).into(), (5000, 5000).into(), tile_size_in_pixels.into())));
+    }
+    #[test]
```

Comment:
This is hard to comprehend.
Why can we load the tiles without specifying a file?

Reply:
So the basic idea is to have Tiles available for all raster datasets. While GDAL knows about tiles, not all raster datasets have them. Most raster files are stride based e.g. the tile information for a 400x400 pixel raster is "400x1". For large rasters a tile based storage like "100x100" is available. And then there are Datasets where Tiles are separated files. To handle this we need something smart, like a TimeInformationProvider. This provider has to supply the tile information. In this test case, this idea is tested. No file is required because the tile information is predefined and independent of the file.